### PR TITLE
fixed #30 Integrate with Sanic

### LINKED ISSUE #30: Integrate with Sanic
Perhaps at least offer a request-scoped lazy-borrowed connection.

### PR DESCRIPTION
**Pull request by lbhsot**

### CODE REVIEW

File: examples/basic.py [hunk 1]
```diff
@@ -16,12 +16,13 @@ def __repr__(self):
 
 
 async def main():
-    conn = await asyncpg.connect('postgresql://localhost/gino')
+    # conn = await asyncpg.connect('postgresql://localhost/gino')
+    conn = await asyncpg.connect(host='localhost', user='lbhsot', password='lbh0625.',database='gino', port=None)
```
Comment (fantix) [inline examples/basic.py]: please double check the diff before PR

File: gino/asyncpg_delegate.py [hunk 2]
```diff
@@ -195,7 +206,7 @@ def get_bind(self, bind=None):
     # noinspection PyUnresolvedReferences
     async def first(self, clause, *multiparams, bind=None,
                     timeout=None, **params):
-        bind = self.get_bind(bind)
+        bind = await self.get_bind(bind)
```
Comment (fantix) [inline gino/asyncpg_delegate.py]: all other `get_bind` calls should be awaited

File: gino/asyncpg_delegate.py [hunk 3]
```diff
@@ -236,7 +247,7 @@ def iterate(self, clause, *multiparams, connection=None,
 
         return rv
 
-    def acquire(self, *, timeout=None, reuse=True):
+    def acquire(self, *, timeout=None, reuse=True, lazy = False):
```
Comment (fantix) [inline gino/asyncpg_delegate.py]: unused, should make use of `LazyTransactionalConnection`? (maybe split transaction away from connection?)

File: gino/sanic_support.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+        @app.middleware('response')
+        async def on_response(request, response):
+            print('on_response')
+            local = get_local()
+            if local:
+                stack = local.get('connection_stack')
+                if stack:
+                    ltc = stack.pop()
+                    await ltc.exit(None if response.status == 200 else True,
```
Comment (fantix) [inline gino/sanic_support.py]: is it confirmed that `HTTP 200` is a good condition for commit or non-rollback?

File: gino/sanic_support.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        if self._ctx is not None:
+            ctx, self._ctx = self._ctx, None
+            await ctx.__aexit__(*args)
+
+
+class SanicMixin:
+    def init_app(self, app):
+        @app.middleware('request')
+        async def on_request(request):
+            print('on_request')
```
Comment (fantix) [inline gino/sanic_support.py]: please remove debug code

File: gino/sanic_support.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            local = get_local()
+            if local:
+                stack = local.get('connection_stack')
+                if stack:
+                    stack.append(LazyTransactionalConnection(self))
+
+        @app.middleware('response')
+        async def on_response(request, response):
+            print('on_response')
```
Comment (fantix) [inline gino/sanic_support.py]: ditto

File: examples/sanic/sanic_basic.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    def __repr__(self):
+        return '{}<{}>'.format(self.nickname, self.id)
+
+
+@app.route("/")
+async def index(request):
+    print(await User.get(6))
+    print(await User.get_or_404(5))
+    return text("Hello World!")
```
Comment (fantix) [inline examples/sanic/sanic_basic.py]: Not quite a good example, yet 😸

File: examples/sanic/sanic_basic.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import asyncpg
+
+from gino import Gino
+from sanic import Sanic
+from sanic.response import text
+from sqlalchemy import Column, BigInteger, Unicode
+
+app = Sanic()
+app.config['GINO_USER'] = 'lbhsot'
```
Comment (fantix) [inline examples/sanic/sanic_basic.py]: debug code, again ... and please use `config.GINO_USER`

File: gino/asyncpg_delegate.py (hunk 3 above)
Comment (fantix) [inline gino/asyncpg_delegate.py]: and there shouldn't be any space around `=` here

**Comment by lbhsot** [comment]
@fantix added some notes.

**Pull request closed without merging**